### PR TITLE
remove use_mock_hardware from moveit launchfile 

### PR DESCRIPTION
**Pull request by fdurchdewald**
The scaled JTC can be used on fake hardware as well. Therefore, there is no real reason for keeping the use_mock_hardware argument in the MoveIt launch file.

### CODE REVIEW

**Comment by fmauch** [comment]
@fdurchdewald Could you please fix  the duplicated target in the `usage.rst` file? Thanks!

File: ur_robot_driver/doc/usage.rst [hunk 1]
```diff
@@ -194,14 +194,18 @@ robot as explained `here <https://moveit.picknik.ai/galactic/doc/tutorials/quick
...
 .. code-block::
 
-   ros2 launch ur_robot_driver ur_control.launch.py ur_type:=ur5e robot_ip:=yyy.yyy.yyy.yyy use_mock_hardware:=true launch_rviz:=false initial_joint_controller:=joint_trajectory_controller
+   ros2 launch ur_robot_driver ur_control.launch.py ur_type:=ur5e robot_ip:=yyy.yyy.yyy.yyy use_mock_hardware:=true launch_rviz:=false
    # and in another shell
-   ros2 launch ur_moveit_config ur_moveit.launch.py ur_type:=ur5e launch_rviz:=true use_mock_hardware:=true
+   ros2 launch ur_moveit_config ur_moveit.launch.py ur_type:=ur5e launch_rviz:=true
+
+Now you should be able to use the MoveIt Plugin in rviz2 to plan and execute trajectories with the
+robot as explained `here <https://moveit.picknik.ai/galactic/doc/tutorials/quickstart_in_rviz/quickstart_in_rviz_tutorial.html>`_.
```
Comment (fmauch) [inline ur_robot_driver/doc/usage.rst:207]:
```suggestion
robot as explained `in the MoveIt! documentation <https://moveit.picknik.ai/galactic/doc/tutorials/quickstart_in_rviz/quickstart_in_rviz_tutorial.html>`_.
```

**Comment by destogl** [comment]
@fdurchdewald does this means that you will have different robot descriptions used on driver launch and MoveIt launch

**Comment by fmauch** [comment]
> @fdurchdewald does this means that you will have different robot descriptions used on driver launch and MoveIt launch

No, we should remove the description from the moveit launch. AFAIK it takes the one from the RSP's description topic by now.

**Comment by fmauch** [comment]
Before merging this, we should consider https://github.com/UniversalRobots/Universal_Robots_ROS2_Gazebo_Simulation/pull/22

What do we do about the scaled jtc in gazebo? Probably we should still provide an option for the default controller.

**Comment by fmauch** [comment]
I think this is now obsolete with #998

**Pull request closed without merging**